### DISCUSSION TITLE
[Connector Request] Impersonator

### DISCUSSION THREAD

**Discussion by VGLoic** [wagmi-dev/references#257] [Connector request]
### What **novel use-case** does the Connector provide?

Hello!

I was wondering if you were interested in an `Impersonator` connector. The goal of this connector would be to let the user choose a valid Ethereum address (through a `prompt` for example) and to navigate the app with this chosen address.

I have a raw version that I will be happy to consolidate in order to make a proposition.

I am using this connector more as as a development or debugging tool but I wonder if some people may consider it even as a feature for production application.

Happy to hear your thoughts on this or if you know similar way to achieve this result :)

### Are the Connector's integrations production-ready and generally available?

No dependencies other than `ethers` are expected.

### Are you committed to actively mainting the Connector?

- [X] Yes, my team is or I am committed to actively maintaining the Connector.

### Additional comments

Thanks for the form, it is well made.

Cheers

**Comment by tmm** [comment]
Thanks for opening this up! Working on migrating wagmi from ethers to [viem](https://viem.sh) at the moment and will circle back soon. Would love to look at some existing code (if any) in the meantime.

**Reply by VGLoic** [reply]
I have a first iteration on [this commit](https://github.com/wagmi-dev/references/commit/cbdaacee2a6681b13b11b62b7ddac0f67ea403d1), it is based on a collection of static JSON RPC provider so it is quite static but quite straightforward in terms of implementation

I am not very familiar with the inner working of the connector so I need to spend a bit of time better understanding which parts need to be implemented 😁 

Good luck for the migration!
